work_queue/src/work_queue_process.c

```diff
@@ -268,4 +269,73 @@ void work_queue_process_kill(struct work_queue_process *p)
 	waitpid(p->pid, NULL, 0);
 }
 
+
+static int download_url(const char *url, const char *cache_name) {
+	debug(D_WQ, "Retrieving %s from (%s)\n", cache_name, url);
+
+	char command[WORK_QUEUE_LINE_MAX];
+	snprintf(command, WORK_QUEUE_LINE_MAX, "curl -f -o \"cache/%s\" \"%s\"", cache_name, url);
+
```

Comment:
There is now a dttools library "shell" which allows you to catch stdout/stderr. You could then output that to the debug log:

https://github.com/cooperative-computing-lab/cctools/blob/55f3b83d3b81cb51e929657699db83f450b06693/dttools/src/shell.h

Here's an example where I use it:

https://github.com/cooperative-computing-lab/cctools/blob/55f3b83d3b81cb51e929657699db83f450b06693/chirp/src/chirp_client.c#L549-L577